solr/core/src/java/org/apache/solr/util/circuitbreaker/CircuitBreakerRegistry.java

```diff
@@ -111,43 +199,82 @@ public static String toErrorMessage(List<CircuitBreaker> circuitBreakerList) {
   }
 
   public boolean isEnabled(SolrRequestType requestType) {
-    return circuitBreakerMap.containsKey(requestType);
+    return circuitBreakerMap.containsKey(requestType)
+        || globalCircuitBreakerMap.containsKey(requestType);
```

Comment:
```suggestion
    return globalCircuitBreakerMap.containsKey(requestType)
        || circuitBreakerMap.containsKey(requestType);
```

Reply:
Why?

Reply:
If (and that is an if) one assumes that global circuit breakers would be more commonly configured (since they are easier to configure) than per-core ones then a successful lookup in the first map would save a lookup attempt in the second map.

Reply:
Hehe. This map will contain two request type enums `QUERY` and `UPDATE`, so they should be super fast anyway. And we don't have reason to believe that one is more used than the other at this point. So I'll leave it as is.